ccmlib/scylla_repository.py

```diff
@@ -234,15 +234,47 @@ def setup(version, verbose=True, skip_downloads=False):
               - release:4.2.1
               - release:2020.1
               - release:2020.1.5
+            flags can be added to the version using [(flag(=value)?)+]
+              - supported flags are:
+                - product=(scylla|scylla-enterprise)
+                - arch=(x86_64|aarch64)
+                - debug
 
     :param verbose: if True, print progress during download
     :param skip_downloads: if True, skips the actual download of files for testing purposes
 
     """
     s3_url = ''
-    type_n_version = version.split(':', 1)
-    scylla_product = os.environ.get('SCYLLA_PRODUCT', 'scylla')
-    scylla_arch = os.environ.get('SCYLLA_ARCH', 'x86_64')
+    version_re = re.compile(r"""
```

Comment:
for such elaborate regex, it would be nice to have a unittest demo the different options are work as expected

Reply:
okay, I tested it manually, but can add a unit test

Reply:
updated tests (and actually had to fix the regex to accommodate for the exiting versions)